protocol/x/vault/keeper/msg_server_deposit_to_vault.go

```diff
@@ -11,5 +15,56 @@ func (k msgServer) DepositToVault(
 	goCtx context.Context,
 	msg *types.MsgDepositToVault,
 ) (*types.MsgDepositToVaultResponse, error) {
+	ctx := lib.UnwrapSDKContext(goCtx, types.ModuleName)
+
+	err := k.Keeper.HandleMsgDepositToVault(ctx, msg)
+	if err != nil {
+		return nil, err
+	}
+
 	return &types.MsgDepositToVaultResponse{}, nil
 }
+
+// HandleMsgDepositToVault handles a MsgDepositToVault.
+func (k Keeper) HandleMsgDepositToVault(
+	ctx sdk.Context,
+	msg *types.MsgDepositToVault,
+) error {
+	// Mint shares for the vault.
+	err := k.MintShares(
+		ctx,
+		*msg.VaultId,
+		msg.SubaccountId.Owner,
+		msg.QuoteQuantums.BigInt(),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Transfer from sender subaccount to vault.
+	// Note: Transfer should take place after minting shares for
+	// shares calculation to be correct.
+	err = k.subaccountsKeeper.TransferFundsFromSubaccountToSubaccount(
```

Comment:
nit: is there any situation in which this errors out?
if so, it could be mean that subaccount could hack us because they keep the money and get shares in the vault.
it doesn't really matter though, since we aren't exposing this to users.

Reply:
transaction is rolled back if it fails so there shouldn't be a scenario where a user gets shares but keeps the money